chalice/config.py

```diff
@@ -8,14 +8,70 @@
 
 
 class Config(object):
-    """Configuration information for a chalice app."""
+    """Configuration information for a chalice app.
+
+    Configuration values for a chalice app can come from
+    a number of locations, files on disk, CLI params, default
+    values, etc.  This object is an abstraction that normalizes
+    these values.
+
+    In general, there's a precedence for looking up
+    config values:
+
+        * User specified params
+        * Config file values
+        * Default values
+
+    A user specified parameter would mean values explicitly
+    specified by a user.  Generally these come from command
+    line parameters (e.g ``--profile prod``), but for the purposes
+    of this object would also mean values passed explicitly to
+    this config object when instantiated.
+
+    Additionally, there are some configurations that can vary
+    per chalice stage (note that a chalice stage is different
+    from an api gateway stage).  For config values loaded from
+    disk, we allow values to be specified for all stages or
+    for a specific stage.  For example, take ``environment_variables``.
+    You can set this as a top level key to specify env vars
+    to set for all stages, or you can set this value per chalice
+    stage to set stage-specific environment variables.  Consider
+    this config file::
+
+        {
+          "environment_variables": {
+            "TABLE": "foo"
+          },
+          "stages": {
+            "dev": {
+              "environment_variables": {
+                "S3BUCKET": "devbucket"
+              }
+            },
+            "prod": {
+              "environment_variables": {
+                "S3BUCKET": "prodbucket",
+                "TABLE": "prodtable"
+              }
+            }
+          }
+        }
+
+    If the currently configured chalice stage is "dev", then
+    the config.environment_variables would be::
+
+        {"TABLE": "foo", "S3BUCKET": "devbucket"}
+
+    """
     def __init__(self,
+                 chalice_stage='dev',
```

Comment:
There is a lot of hardcoded default ``dev`` values. Wonder if it makes sense to have it be a constant to import or required instead of optional in case it ever needs to be updated?